File: samples/client/3_1_0_unit_test/java/src/test/java/org/openapijsonschematools/client/components/schemas/ContainsKeywordValidationTest.java
package org.openapijsonschematools.client.components.schemas;

import org.junit.Assert;
import org.junit.Test;
import org.openapijsonschematools.client.configurations.JsonSchemaKeywordFlags;
import org.openapijsonschematools.client.configurations.SchemaConfiguration;
import org.openapijsonschematools.client.exceptions.ValidationException;
import org.openapijsonschematools.client.exceptions.InvalidTypeException;
import org.openapijsonschematools.client.schemas.validation.MapUtils;
import org.checkerframework.checker.nullness.qual.Nullable;

import java.util.Arrays;
import java.util.List;
import java.util.Map;
import java.util.AbstractMap;

public class ContainsKeywordValidationTest {
    static final SchemaConfiguration configuration = new SchemaConfiguration(JsonSchemaKeywordFlags.ofNone());

    @Test
    public void testArrayWithTwoItemsMatchingSchema56IsValidPasses() {
        // array with two items matching schema (5, 6) is valid
        final var schema = ContainsKeywordValidation.ContainsKeywordValidation1.getInstance();
        schema.validate(
            Arrays.asList(
                3,
                4,
                5,
                6
            ),
            configuration
        );
    }

    @Test
    public void testNotArrayIsValidPasses() {
        // not array is valid
        final var schema = ContainsKeywordValidation.ContainsKeywordValidation1.getInstance();
        schema.validate(
            MapUtils.makeMap(
            ),
            configuration
        );
    }

    @Test
    public void testArrayWithItemMatchingSchema5IsValidPasses() {
        // array with item matching schema (5) is valid
        final var schema = ContainsKeywordValidation.ContainsKeywordValidation1.getInstance();
        schema.validate(
            Arrays.asList(
                3,
                4,
                5
            ),
            configuration
        );
    }

    @Test
    public void testArrayWithItemMatchingSchema6IsValidPasses() {
        // array with item matching schema (6) is valid
        final var schema = ContainsKeywordValidation.ContainsKeywordValidation1.getInstance();
        schema.validate(
            Arrays.asList(
                3,
                4,
                6
            ),
            configuration
        );
    }

    @Test
    public void testArrayWithoutItemsMatchingSchemaIsInvalidFails() {
        // array without items matching schema is invalid
        final var schema = ContainsKeywordValidation.ContainsKeywordValidation1.getInstance();
        try {
            schema.validate(
                Arrays.asList(
                    2,
                    3,
                    4
                ),
                configuration
            );
            throw new RuntimeException("A different exception must be thrown");
        } catch (ValidationException | InvalidTypeException ignored) {
            ;
        }
    }

    @Test
    public void testEmptyArrayIsInvalidFails() {
        // empty array is invalid
        final var schema = ContainsKeywordValidation.ContainsKeywordValidation1.getInstance();
        try {
            schema.validate(
                Arrays.asList(
                ),
                configuration
            );
            throw new RuntimeException("A different exception must be thrown");
        } catch (ValidationException | InvalidTypeException ignored) {
            ;
        }
    }
}

File: samples/client/3_1_0_unit_test/kotlin/src/test/kotlin/org/openapijsonschematools/client/components/schemas/ContainsKeywordValidationTest.kt
package org.openapijsonschematools.client.components.schemas

import org.openapijsonschematools.client.configurations.JsonSchemaKeyword
import org.openapijsonschematools.client.configurations.SchemaConfiguration
import org.openapijsonschematools.client.exceptions.ValidationException

import kotlin.test.Test
import kotlin.test.assertFailsWith

class ContainsKeywordValidationTest {
    companion object {
        val configuration = SchemaConfiguration(disabledKeywordFlags=setOf(JsonSchemaKeyword.FORMAT))
    }

    @Test
    fun testArrayWithTwoItemsMatchingSchema56IsValidPasses() {
        // array with two items matching schema (5, 6) is valid
        val schema = ContainsKeywordValidation.ContainsKeywordValidation1.getInstance()
        schema.validate(
            listOf(
                3,
                4,
                5,
                6
            ),
            configuration
        )
    }

    @Test
    fun testNotArrayIsValidPasses() {
        // not array is valid
        val schema = ContainsKeywordValidation.ContainsKeywordValidation1.getInstance()
        schema.validate(
            mapOf<Any?, Any?>(
            ),
            configuration
        )
    }

    @Test
    fun testArrayWithItemMatchingSchema5IsValidPasses() {
        // array with item matching schema (5) is valid
        val schema = ContainsKeywordValidation.ContainsKeywordValidation1.getInstance()
        schema.validate(
            listOf(
                3,
                4,
                5
            ),
            configuration
        )
    }

    @Test
    fun testArrayWithItemMatchingSchema6IsValidPasses() {
        // array with item matching schema (6) is valid
        val schema = ContainsKeywordValidation.ContainsKeywordValidation1.getInstance()
        schema.validate(
            listOf(
                3,
                4,
                6
            ),
            configuration
        )
    }

    @Test
    fun testArrayWithoutItemsMatchingSchemaIsInvalidFails() {
        // array without items matching schema is invalid
        val schema = ContainsKeywordValidation.ContainsKeywordValidation1.getInstance()
        assertFailsWith<ValidationException>(
            block = {
                schema.validate(
                    listOf(
                        2,
                        3,
                        4
                    ),
                    configuration
                )
            }
        )
    }

    @Test
    fun testEmptyArrayIsInvalidFails() {
        // empty array is invalid
        val schema = ContainsKeywordValidation.ContainsKeywordValidation1.getInstance()
        assertFailsWith<ValidationException>(
            block = {
                schema.validate(
                    listOf<Any?>(
                    ),
                    configuration
                )
            }
        )
    }
}
